File: Front-End/src/services/brand.service.ts
import axiosClient from '@/configurations/axios.config';
import type {
  CreateBrandRequest,
  BrandResponse,
  BrandListResponse
} from '@/types/brand.type';

export const brandService = {
  getBrands: async (page: number = 1, size: number = 10, search: string = "") => {
    const response = await axiosClient.get<BrandListResponse>(`/brands?page=${page}&size=${size}&brandName=${search}`);
    return response.data;
  },

  getBrandById: async (id: number) => {
    const response = await axiosClient.get<BrandResponse>(`/brands/${id}`);
    return response.data;
  },

  createBrand: async (request: CreateBrandRequest) => {
    const response = await axiosClient.post<BrandResponse>('/brands', request);
    return response.data;
  },

  updateBrand: async (id: number, request: CreateBrandRequest) => {
    const response = await axiosClient.put<BrandResponse>(`/brands/${id}`, request);
    return response.data;
  },

  changeStatusBrand: async (id: number) => {
    await axiosClient.put(`/brands/change-status/${id}`);
  }
};

File: Back-End/src/main/java/iuh/fit/ecommerce/services/BrandService.java
package iuh.fit.ecommerce.services;

import iuh.fit.ecommerce.dtos.request.brand.BrandAddRequest;
import iuh.fit.ecommerce.dtos.response.base.ResponseWithPagination;
import iuh.fit.ecommerce.dtos.response.brand.BrandResponse;
import iuh.fit.ecommerce.entities.Brand;

import java.util.List;

public interface BrandService {

    BrandResponse createBrand(BrandAddRequest request);

    ResponseWithPagination<List<BrandResponse>> getBrands(int page, int size, String brandName);

    BrandResponse getBrandById(Long id);

    BrandResponse updateBrand(Long id, BrandAddRequest request);

    void changeStatusBrand(Long id);
    Brand getBrandEntityById(Long id);
}
